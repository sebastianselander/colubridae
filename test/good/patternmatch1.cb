// declare an algebraic data type
type Foo {
    Bar,
    Foo,
    Baz(int, Foo),
}

def main() {
    let var = Baz(420, Baz(69, Bar));
    // pattern match
    let x = match var {
        Baz(n,m) => {
            printInt(n); 
            printString("\n");
            match m {
                Baz(k, x) => {
                    printInt(k);
                    printString("\n");
                    match x {
                        Bar => printString("got bar\n")
                    }
                }
            }
        }
    };
}
