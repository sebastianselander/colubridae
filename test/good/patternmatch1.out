420
69
got bar
